doc/data_model.md

```diff
@@ -0,0 +1,95 @@
+# Data model
+
+## Overall
+
+The canonical model closely follows the [TTML 2](https://www.w3.org/TR/ttml2) data model, as constrained by the [IMSC 1.1 Text Profile](https://www.w3.org/TR/ttml-imsc1.1/#text-profile) specification. This includes both the static structure of the model as well as temporal, layout and style processing. The objective is for a valid IMSC 1.1 Text Profile document to be mapped into a canonical model instance such that presenting instance results in the same outout as the input IMSC document.
+
+The canonical model is specified in `ttconv.model`.
+
+The class hierarchy of the canonical model is summarized the following figure:
+
+```txt
+  Document
+    : Region* Body?
+
+  Body
+    : Div*
+  
+  Div
+    : (P | Div)*
+  
+  P
+    : (Span | Ruby | Br)*
+  
+  Span
+    : (Span | Br | Text)*
+  
+  Ruby
+    : Rb? Rt?
+    | Rb? Rp Rt? Rp
+    | Rbc Rtc Rtc?
+  
+  Rbc
+    : Rb*
+  
+  Rtc
+  : Rt*
+  | Rp Rt* Rp
+
+  Rb, Rt, Rp
+    : Span*
+```
+
+where:
+
+* the `Document` class corresponds to the `tt` element
+* the `Body`, `Div`, `P`, `Span`, `Br` and `Region` classes corresponds to the TTML content element of the same name, respectively
+* the `Text` class corresponds to a TTML text node
+* the `Ruby`, `Rt`, `Rb`, `Rtc`, `Rbc`, `Rp` classes correspond to the TTML `span` element with the computed value of `tts:ruby` attribute specified in the following table
+
+| Canonical model class | Computed value of `tts:ruby` attribute |
+|-----------------------|----------------------------------------|
+| `Ruby`                | `container`                            |
+| `Rt`                  | `text`                                 |
+| `Rb`                  | `base`                                 |
+| `Rtc`                 | `textContainer`                        |
+| `Rbc`                 | `baseContainer`                        |
+| `Rp`                  | `delimiter`                            |
+
+## Basic operation
+
+The canonical model allows content elements (instances of `ttconv.model.ContentElement`) to be arranged in a hierarchical structures (using the `ttconv.model.ContentElement.add_child()` and `ttconv.model.ContentElement.add_child()`) that are associated with a single document (using the `ttconv.model.ContentElement.set_doc()` method with an instance of `ttconv.model.Document`).
```

Comment:
Shouldn't it be `ttconv.model.ContentElement.push_child()` and `ttconv.model.ContentElement.remove_child()` instead of twice `ttconv.model.ContentElement.add_child()`?

Reply:
Fixed